src/kOS.Safe/Binding/BoundVariable.cs

```diff
@@ -20,6 +22,8 @@ public override object Value
                 // new primitive encapsulation types we instead encapsulate any value returned
                 // by the delegate.  This makes it so that all of the getters for bound variables
                 // don't need to be modified to explicitly return the encapsulated types.
+                if (Volatile)
+                    return Structure.FromPrimitive(Get());
                 return currentValue ?? (currentValue = Structure.FromPrimitive(Get()));
```

Comment:
I always have a preference for not duplicating a call if there's a way to write it only mentioning the call once. (less likely for a future edit to forget one of the two duplicated places).

how about:
```
if (!Volatile && currentValue != null)
    return currentValue;
return (currentValue = Structure.FromPrimitive(Get()));
```